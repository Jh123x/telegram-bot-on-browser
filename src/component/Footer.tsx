import * as React from "react";
import Container from "@mui/material/Container";
import Typography from "@mui/material/Typography";
import Link from "@mui/material/Link";
import { Box } from "@mui/material";
import styles from "./Footer.module.css";

export const Footer = () => {
  return (
    <Box className={styles.footer} component="footer">
      <Container maxWidth="sm">
        <Typography variant="body2" className={styles.text_box}>
          {"Copyright Â©"}
          <Link
            color="inherit"
            href="https://jh123x.com/"
            className={styles.url}
          >
            Jh123x
          </Link>
          {new Date().getFullYear()}
        </Typography>
      </Container>
    </Box>
  );
};
