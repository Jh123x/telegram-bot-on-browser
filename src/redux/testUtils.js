import React from "react";
import { render } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import { botSlice, defaultBotState } from "./botSlice";

export const generateDefaultState = () => {
  return {
    bot: defaultBotState,
  };
};

export const setupStore = (preloadedState) => {
  return configureStore({
    reducer: {
      bot: botSlice.reducer,
    },
    preloadedState,
  });
};

export function renderWithProviders(
  ui,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({
      reducer: { bot: botSlice.reducer },
      preloadedState: preloadedState,
    }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
