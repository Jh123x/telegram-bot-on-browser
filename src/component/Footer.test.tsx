import { test, expect } from "@jest/globals";
import React from "react";
import { generateDefaultState, renderWithProviders, setupStore } from "../redux/testUtils.tsx";
import { Footer } from "./Footer.tsx";

test("renders footer correctly", () => {
    const store = setupStore(generateDefaultState());
    const component = renderWithProviders(<Footer />, { store: store });
    expect(component).toMatchSnapshot();
});
