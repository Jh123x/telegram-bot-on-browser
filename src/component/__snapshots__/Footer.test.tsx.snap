// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders footer correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="footer MuiBox-root css-0"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthSm css-zzzaw2-MuiContainer-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 text_box css-7ctihp-MuiTypography-root"
          >
            Copyright ©
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways url css-drt9hv-MuiTypography-root-MuiLink-root"
              href="https://jh123x.com/"
            >
              Jh123x
            </a>
            2025
          </p>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="footer MuiBox-root css-0"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthSm css-zzzaw2-MuiContainer-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 text_box css-7ctihp-MuiTypography-root"
        >
          Copyright ©
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways url css-drt9hv-MuiTypography-root-MuiLink-root"
            href="https://jh123x.com/"
          >
            Jh123x
          </a>
          2025
        </p>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
